spyder/plugins/editor/widgets/codeeditor/lsp_mixin.py

```diff
@@ -1199,6 +1204,9 @@ def _apply_document_edits(self, edits):
 
             # End text insertion
             cursor.endEditBlock()
+            
+            if breakpoints is not None:
```

Comment:
```suggestion
            # Restore breakpoints
            if breakpoints is not None and breakpoints:
```

This also validates that `breakpoints` is not empty before proceeding.